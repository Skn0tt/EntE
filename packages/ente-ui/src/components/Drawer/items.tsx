/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";
import { List, ListItem, ListItemIcon, ListItemText } from "@material-ui/core";
import { WatchLater, InsertDriveFile, Person } from "@material-ui/icons";
import { Route } from "react-router-dom";

/**
 * # Items
 */
const Slots: React.SFC = () => (
  <Route
    render={({ history }) => (
      <ListItem button onClick={() => history.push("/slots")}>
        <ListItemIcon>
          <WatchLater />
        </ListItemIcon>
        <ListItemText primary="Stunden" />
      </ListItem>
    )}
  />
);

const Users: React.SFC = () => (
  <Route
    render={({ history }) => (
      <ListItem button onClick={() => history.push("/users")}>
        <ListItemIcon>
          <Person />
        </ListItemIcon>
        <ListItemText primary="Nutzer" />
      </ListItem>
    )}
  />
);

const Entries: React.SFC = () => (
  <Route
    render={({ history }) => (
      <ListItem button onClick={() => history.push("/entries")}>
        <ListItemIcon>
          <InsertDriveFile />
        </ListItemIcon>
        <ListItemText primary="EintrÃ¤ge" />
      </ListItem>
    )}
  />
);

/**
 * # Composed
 */
export const AdminItems: React.SFC = () => (
  <List>
    <Entries />
    <Slots />
    <Users />
  </List>
);

export const TeacherItems: React.SFC = () => (
  <List>
    <Slots />
  </List>
);

export const StudentItems: React.SFC = () => (
  <List>
    <Entries />
  </List>
);

export const ParentItems: React.SFC = () => (
  <List>
    <Entries />
  </List>
);

export const ManagerItems: React.SFC = () => (
  <List>
    <Entries />
    <Slots />
  </List>
);
