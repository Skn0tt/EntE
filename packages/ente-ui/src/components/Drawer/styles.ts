/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import {
  StyleRules,
  StyleRulesCallback
} from "@material-ui/core/styles/withStyles";
import { Theme } from "@material-ui/core/styles/createMuiTheme";

const drawerWidth = 240;

const styles: StyleRulesCallback = theme => ({
  root: {
    width: "100%",
    zIndex: 1,
    overflowX: "hidden"
  },
  appFrame: {
    position: "relative",
    display: "flex",
    width: "100%",
    height: "100%"
  },
  appBar: {
    position: "fixed",
    marginLeft: drawerWidth,
    [theme.breakpoints.up("md")]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  grow: {
    flex: "1 1 auto"
  },
  logo: {
    display: "block",
    [theme.breakpoints.down("sm")]: {
      position: "fixed",
      left: "50%",
      marginLeft: "-62px"
    }
  },
  navIconHide: {
    [theme.breakpoints.up("md")]: {
      display: "none"
    },
    color: "white"
  },
  drawerPaper: {
    width: 250,
    [theme.breakpoints.up("md")]: {
      width: drawerWidth,
      position: "fixed",
      height: "100%"
    }
  },
  content: {
    width: "100%",
    marginTop: 64,
    [theme.breakpoints.up("md")]: {
      marginLeft: drawerWidth
    }
  },
  loadingIndicator: {
    position: "fixed",
    width: "100%"
  }
});

export default styles;
