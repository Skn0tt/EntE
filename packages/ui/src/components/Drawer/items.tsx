import * as React from "react";
import { List, ListItem, ListItemIcon, ListItemText } from "material-ui";
import { WatchLater, InsertDriveFile, Person } from "material-ui-icons";
import { Route } from "react-router-dom";

/**
 * # Items
 */
const Slots: React.SFC = () => (
  <Route
    render={({ history }) => (
      <ListItem button onClick={() => history.push("/slots")}>
        <ListItemIcon>
          <WatchLater />
        </ListItemIcon>
        <ListItemText primary="Stunden" />
      </ListItem>
    )}
  />
);

const Users: React.SFC = () => (
  <Route
    render={({ history }) => (
      <ListItem button onClick={() => history.push("/users")}>
        <ListItemIcon>
          <Person />
        </ListItemIcon>
        <ListItemText primary="Nutzer" />
      </ListItem>
    )}
  />
);

const Entries: React.SFC = () => (
  <Route
    render={({ history }) => (
      <ListItem button onClick={() => history.push("/entries")}>
        <ListItemIcon>
          <InsertDriveFile />
        </ListItemIcon>
        <ListItemText primary="EintrÃ¤ge" />
      </ListItem>
    )}
  />
);

/**
 * # Composed
 */
export const AdminItems: React.SFC = () => (
  <List>
    <Entries />
    <Slots />
    <Users />
  </List>
);

export const TeacherItems: React.SFC = () => (
  <List>
    <Slots />
  </List>
);

export const StudentItems: React.SFC = () => (
  <List>
    <Entries />
  </List>
);

export const ParentItems: React.SFC = () => (
  <List>
    <Entries />
  </List>
);

export const ManagerItems: React.SFC = () => (
  <List>
    <Entries />
    <Slots />
  </List>
);
