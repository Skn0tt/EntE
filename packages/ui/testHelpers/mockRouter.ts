import { RouteComponentProps } from "react-router";
import { Location, UnregisterCallback } from "history";

export function getMockRouterProps<P>(data: P) {
  const location: Location = {
    hash: "",
    key: "",
    pathname: "",
    search: "",
    state: {}
  };

  const props: RouteComponentProps<P> = {
    location,
    match: {
      isExact: true,
      params: data,
      path: "",
      url: ""
    },
    history: {
      location,
      length: 2,
      action: "POP",
      push: () => {},
      replace: () => {},
      go: num => {},
      goBack: () => {},
      goForward: () => {},
      block: t => null as any,
      createHref: t => "",
      listen: t => null as any
    },
    staticContext: {}
  };

  return props;
}
