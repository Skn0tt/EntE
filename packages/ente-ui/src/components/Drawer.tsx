/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import { AppState, getRole, isLoading } from "../redux";
import { Roles } from "ente-types";
import {
  AppBar,
  Hidden,
  IconButton,
  Toolbar,
  Drawer as MUIDrawer,
  List,
  Divider,
  Theme
} from "@material-ui/core";
import * as React from "react";
import MenuIcon from "@material-ui/icons/Menu";
import { connect } from "react-redux";
import LoginStatus from "./LoginStatus";
import RefreshButton from "./RefreshButton";
import {
  AdminItems,
  ManagerItems,
  ParentItems,
  StudentItems,
  TeacherItems
} from "./Drawer.items";
import { Maybe } from "monet";
import SettingsMenu from "./SettingsMenu";
import { useToggle } from "../helpers/useToggle";
import { makeStyles } from "@material-ui/styles";

const drawerWidth = 240;

const useStyles = makeStyles(
  (theme: Theme) => ({
    root: {
      width: "100%",
      zIndex: 1,
      overflowX: "hidden"
    },
    appFrame: {
      position: "relative",
      display: "flex",
      width: "100%",
      height: "100%"
    },
    appBar: {
      position: "fixed",
      marginLeft: drawerWidth,
      [theme.breakpoints.up("md")]: {
        width: `calc(100% - ${drawerWidth}px)`
      }
    },
    toolBar: {
      paddingRight: 0
    },
    grow: {
      flex: "1 1 auto"
    },
    logo: {
      display: "block",
      [theme.breakpoints.down("sm")]: {
        position: "fixed",
        left: "50%",
        marginLeft: "-62px"
      }
    },
    navIconHide: {
      [theme.breakpoints.up("md")]: {
        display: "none"
      },
      color: "white"
    },
    drawerPaper: {
      width: 250,
      [theme.breakpoints.up("md")]: {
        width: drawerWidth,
        position: "fixed",
        height: "100%"
      }
    },
    content: {
      width: "100%",
      minHeight: "calc(100vh - 56px)",
      marginTop: 56,
      backgroundColor: theme.palette.background.default,
      [theme.breakpoints.up("md")]: {
        height: "calc(100vh - 64px)",
        marginTop: 64,
        marginLeft: drawerWidth
      }
    }
  }),
  { withTheme: true }
);

interface DrawerOwnProps {}
interface DrawerStateProps {
  loading: boolean;
  role: Maybe<Roles>;
}
const mapStateToProps = (state: AppState): DrawerStateProps => ({
  loading: isLoading(state),
  role: getRole(state)
});

type DrawerProps = DrawerOwnProps & DrawerStateProps;

const Drawer: React.FunctionComponent<DrawerProps> = props => {
  const { role, children } = props;
  const [mobileOpen, toggleMobileOpen] = useToggle(false);
  const classes = useStyles(props);

  const drawer = (
    <List disablePadding>
      <LoginStatus />
      <Divider />
      {
        {
          [Roles.ADMIN]: <AdminItems />,
          [Roles.TEACHER]: <TeacherItems />,
          [Roles.PARENT]: <ParentItems />,
          [Roles.STUDENT]: <StudentItems />,
          [Roles.MANAGER]: <ManagerItems />
        }[role.some()]
      }
    </List>
  );

  return (
    <div className={classes.root}>
      <div className={classes.appFrame}>
        <AppBar className={classes.appBar}>
          <Toolbar className={classes.toolBar}>
            <IconButton
              aria-label="open drawer"
              onClick={toggleMobileOpen}
              className={classes.navIconHide}
            >
              <MenuIcon />
            </IconButton>
            <img
              className={classes.logo}
              src={require("../assets/Logo.svg")}
              height={24}
            />
            <div className={classes.grow} />

            <RefreshButton />
            <SettingsMenu />
          </Toolbar>
        </AppBar>
        <Hidden mdUp>
          <MUIDrawer
            variant="temporary"
            open={mobileOpen}
            classes={{
              paper: classes.drawerPaper
            }}
            onClose={toggleMobileOpen}
            ModalProps={{
              keepMounted: true
            }}
          >
            {drawer}
          </MUIDrawer>
        </Hidden>
        <Hidden smDown implementation="css">
          <MUIDrawer
            variant="permanent"
            open
            classes={{
              paper: classes.drawerPaper
            }}
          >
            {drawer}
          </MUIDrawer>
        </Hidden>
        <main className={classes.content}>{children}</main>
      </div>
    </div>
  );
};

export default connect<DrawerStateProps, {}, DrawerOwnProps, AppState>(
  mapStateToProps
)(Drawer);
