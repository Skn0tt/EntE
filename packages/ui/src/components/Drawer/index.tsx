import { AppState, getRole, isLoading } from "ente-redux";
import { Roles } from "ente-types";
import {
  AppBar,
  Divider,
  Hidden,
  IconButton,
  Toolbar,
  WithStyles,
  withStyles,
  Drawer as MUIDrawer
} from "material-ui";
import * as React from "react";
import { Menu as MenuIcon } from "material-ui-icons";
import { connect } from "react-redux";
import { RouteComponentProps, withRouter } from "react-router";
import LoginStatus from "../LoginStatus";
import RefreshButton from "../RefreshButton";
import {
  AdminItems,
  ManagerItems,
  ParentItems,
  StudentItems,
  TeacherItems
} from "./items";
import styles from "./styles";

/**
 * # Component Types
 */
interface StateProps {
  loading: boolean;
  role: Roles;
}
const mapStateToProps = (state: AppState) => ({
  loading: isLoading(state),
  role: getRole(state)
});

type Props = StateProps & RouteComponentProps<{}> & WithStyles;
interface State {
  mobileOpen: boolean;
}

/**
 * # Component Logic
 */
export const toggleDrawer = (s: State): State => ({
  mobileOpen: !s.mobileOpen
});

/**
 * # Component
 */
class Drawer extends React.Component<Props, State> {
  state = {
    mobileOpen: false
  };

  handleDrawerToggle = () => this.setState(toggleDrawer);

  render() {
    const { classes, role, children } = this.props;

    const drawer = (
      <>
        <LoginStatus />
        <Divider />
        {
          {
            [Roles.ADMIN]: <AdminItems />,
            [Roles.TEACHER]: <TeacherItems />,
            [Roles.PARENT]: <ParentItems />,
            [Roles.STUDENT]: <StudentItems />,
            [Roles.MANAGER]: <ManagerItems />
          }[role]
        }
      </>
    );

    return (
      <div className={classes.root}>
        <div className={classes.appFrame}>
          <AppBar className={classes.appBar}>
            <Toolbar>
              <IconButton
                aria-label="open drawer"
                onClick={this.handleDrawerToggle}
                className={classes.navIconHide}
              >
                <MenuIcon />
              </IconButton>
              <img
                className={classes.logo}
                src={require("../../res/img/Logo.svg")}
                height={24}
              />
              <div className={classes.grow} />
              <RefreshButton />
            </Toolbar>
          </AppBar>
          <Hidden mdUp>
            <MUIDrawer
              variant="temporary"
              open={mobileOpen}
              classes={{
                paper: classes.drawerPaper
              }}
              onClose={this.handleDrawerToggle}
              ModalProps={{
                keepMounted: true
              }}
            >
              {drawer}
            </MUIDrawer>
          </Hidden>
          <Hidden smDown implementation="css">
            <MUIDrawer
              variant="permanent"
              open
              classes={{
                paper: classes.drawerPaper
              }}
            >
              {drawer}
            </MUIDrawer>
          </Hidden>
          <main className={classes.content}>{children}</main>
        </div>
      </div>
    );
  }
}

export default withRouter(connect(mapStateToProps)(withStyles(styles)(Drawer)));
