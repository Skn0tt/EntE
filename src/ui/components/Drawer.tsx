/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import { AppState, getRole, isLoading, isAdmin } from "../redux";
import { Roles } from "@@types";
import {
  AppBar,
  Hidden,
  IconButton,
  Toolbar,
  Drawer as MUIDrawer,
  List,
  Divider,
  Theme,
} from "@material-ui/core";
import * as React from "react";
import MenuIcon from "@material-ui/icons/Menu";
import { connect } from "react-redux";
import LoginStatus from "./LoginStatus";
import RefreshButton from "./RefreshButton";
import {
  AdminItems,
  ManagerItems,
  ParentItems,
  StudentItems,
  TeacherItems,
} from "./Drawer.items";
import { Maybe } from "monet";
import SettingsMenu from "./SettingsMenu";
import { useToggle } from "../helpers/useToggle";
import { makeStyles } from "@material-ui/styles";
import Logo from "../assets/Logo.svg";

interface DrawerStateContextType {
  open: boolean;
  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
}

const DrawerStateContext = React.createContext<DrawerStateContextType>({
  open: false,
  setOpen: () => {},
});

function useDrawerState(): [
  boolean,
  React.Dispatch<React.SetStateAction<boolean>>
] {
  const { open, setOpen } = React.useContext(DrawerStateContext);
  return [open, setOpen];
}

export function DrawerStateProvider(props: React.PropsWithChildren<{}>) {
  const [open, setOpen] = React.useState(false);
  const value = React.useMemo(
    () => ({
      open,
      setOpen,
    }),
    [open, setOpen]
  );

  return (
    <DrawerStateContext.Provider value={value}>
      {props.children}
    </DrawerStateContext.Provider>
  );
}

const drawerWidth = 240;

const useStyles = makeStyles(
  (theme: Theme) => ({
    root: {
      width: "100%",
      zIndex: 1,
      overflowX: "hidden",
    },
    appFrame: {
      position: "relative",
      display: "flex",
      width: "100%",
      height: "100%",
    },
    appBar: {
      position: "fixed",
      marginLeft: drawerWidth,
      [theme.breakpoints.up("md")]: {
        width: `calc(100% - ${drawerWidth}px)`,
      },
    },
    toolBar: {
      paddingRight: 0,
    },
    grow: {
      flex: "1 1 auto",
    },
    logo: {
      display: "block",
      [theme.breakpoints.down("sm")]: {
        position: "fixed",
        left: "50%",
        marginLeft: "-62px",
      },
    },
    navIconHide: {
      [theme.breakpoints.up("md")]: {
        display: "none",
      },
      color: "white",
    },
    drawerPaper: {
      width: 250,
      [theme.breakpoints.up("md")]: {
        width: drawerWidth,
        position: "fixed",
        height: "100%",
      },
    },
    content: {
      width: "100%",
      minHeight: "calc(100vh - 56px)",
      marginTop: 56,
      backgroundColor: theme.palette.background.default,
      [theme.breakpoints.up("md")]: {
        height: "calc(100vh - 64px)",
        marginTop: 64,
        marginLeft: drawerWidth,
      },
    },
  }),
  { withTheme: true }
);

interface DrawerOwnProps {}
interface DrawerStateProps {
  loading: boolean;
  role: Maybe<Roles>;
  isAdmin: boolean;
}
const mapStateToProps = (state: AppState): DrawerStateProps => ({
  loading: isLoading(state),
  role: getRole(state),
  isAdmin: isAdmin(state),
});

type DrawerProps = DrawerOwnProps & DrawerStateProps;

const Drawer: React.FunctionComponent<DrawerProps> = (props) => {
  const { role, children, isAdmin } = props;
  const [mobileOpen, setMobileOpen] = useDrawerState();
  const classes = useStyles(props);

  const toggleMobileOpen = React.useCallback(() => {
    setMobileOpen((v) => !v);
  }, [setMobileOpen]);

  const drawer = (
    <List disablePadding>
      <LoginStatus />
      <Divider />
      {
        {
          [Roles.TEACHER]: <TeacherItems />,
          [Roles.PARENT]: <ParentItems />,
          [Roles.STUDENT]: <StudentItems />,
          [Roles.MANAGER]: <ManagerItems />,
        }[role.orSome(Roles.PARENT)]
      }
      {isAdmin && <AdminItems />}
    </List>
  );

  return (
    <div className={classes.root}>
      <div className={classes.appFrame}>
        <AppBar className={classes.appBar}>
          <Toolbar className={classes.toolBar}>
            <IconButton
              aria-label="open drawer"
              onClick={toggleMobileOpen}
              className={classes.navIconHide}
            >
              <MenuIcon />
            </IconButton>
            <Logo height={24} className={classes.logo} />
            <div className={classes.grow} />

            <RefreshButton />
            <SettingsMenu />
          </Toolbar>
        </AppBar>
        <Hidden mdUp>
          <MUIDrawer
            variant="temporary"
            open={mobileOpen}
            classes={{
              paper: classes.drawerPaper,
            }}
            onClose={toggleMobileOpen}
            ModalProps={{
              keepMounted: true,
            }}
          >
            {drawer}
          </MUIDrawer>
        </Hidden>
        <Hidden smDown implementation="css">
          <MUIDrawer
            variant="permanent"
            open
            classes={{
              paper: classes.drawerPaper,
            }}
          >
            {drawer}
          </MUIDrawer>
        </Hidden>
        <main className={classes.content}>{children}</main>
      </div>
    </div>
  );
};

export default connect<DrawerStateProps, {}, DrawerOwnProps, AppState>(
  mapStateToProps
)(Drawer);
