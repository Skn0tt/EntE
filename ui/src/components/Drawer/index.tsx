import * as React from 'react';
import {
  AppBar,
  Toolbar,
  IconButton,
  Typography,
  Hidden,
  Divider,
  withStyles,
  Drawer as MUIDrawer,
  WithStyles,
} from 'material-ui';
import { Menu as MenuIcon } from 'material-ui-icons';
import LoadingIndicator from '../LoadingIndicator';
import * as select from '../../redux/selectors';

import styles from './styles';
import {
  AdminItems,
  ManagerItems,
  TeacherItems,
  StudentItems,
  ParentItems,
} from './items';
import RefreshButton from '../RefreshButton';
import { connect } from 'react-redux';
import { AppState, Roles } from '../../interfaces/index';
import { withRouter, RouteComponentProps } from 'react-router';
import LoginStatus from '../LoginStatus';

interface OwnProps {}

interface StateProps {
  loading: boolean;
  role: Roles;
}

interface State {
  mobileOpen: boolean;
}

const mapStateToProps = (state: AppState) => ({
  loading: select.isLoading(state),
  role: select.getRole(state),
});

type Props = OwnProps & StateProps & RouteComponentProps<{}> & WithStyles<string>;

const Drawer =
  withRouter(
  connect<StateProps, null, OwnProps>(mapStateToProps)(
  withStyles(styles)(
class extends React.Component<Props, State> {
  state = {
    mobileOpen: false,
  };

  handleDrawerToggle = () => this.setState({ mobileOpen: !this.state.mobileOpen });

  render() {
    const { classes } = this.props;

    const drawer = (
      <div>
        <LoginStatus />
        <Divider />
        {this.props.role === Roles.ADMIN && <AdminItems />}
        {this.props.role === Roles.TEACHER && <TeacherItems />}
        {this.props.role === Roles.PARENT && <ParentItems />}
        {this.props.role === Roles.STUDENT && <StudentItems />}
        {this.props.role === Roles.MANAGER && <ManagerItems />}
      </div>
    );

    return (
      <div className={classes.root}>
        {this.props.loading && <LoadingIndicator />}
        <div className={classes.appFrame}>
          <AppBar className={classes.appBar}>
            <Toolbar>
              <IconButton
                aria-label="open drawer"
                onClick={this.handleDrawerToggle}
                className={classes.navIconHide}
              >
                <MenuIcon />
              </IconButton>
              <Typography variant="title" color="inherit" noWrap>
                EntschuldigungsVerfahren
              </Typography>
              <div className={classes.grow} />
              <RefreshButton />
            </Toolbar>
          </AppBar>
          <Hidden mdUp>
            <MUIDrawer
              variant="temporary"
              open={this.state.mobileOpen}
              classes={{
                paper: classes.drawerPaper,
              }}
              onClose={this.handleDrawerToggle}
              ModalProps={{
                keepMounted: true,
              }}
            >
              {drawer}
            </MUIDrawer>
          </Hidden>
          <Hidden smDown implementation="css">
            <MUIDrawer
              variant="permanent"
              open
              classes={{
                paper: classes.drawerPaper,
              }}
            >
              {drawer}
            </MUIDrawer>
          </Hidden>
          <main className={classes.content}>
            {this.props.children}
          </main>
        </div>
      </div>
    );
  }
})));

export default Drawer;
