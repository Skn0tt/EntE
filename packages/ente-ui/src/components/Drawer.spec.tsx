/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";
import Drawer from "./Drawer";
import { shallow } from "enzyme";
import { getMockRouterProps } from "../../testHelpers/mockRouter";

describe("Drawer", () => {
  it("renders correctly", () => {
    const comp = shallow(
      <Drawer {...getMockRouterProps<{}>({})}>Hallo</Drawer>
    );
    expect(comp).toMatchSnapshot();
  });
});
