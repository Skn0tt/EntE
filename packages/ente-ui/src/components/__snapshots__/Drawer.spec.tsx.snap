// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(withRouter(Connect(Drawer)))
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 2,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "",
          "pathname": "",
          "search": "",
          "state": Object {},
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    location={
      Object {
        "hash": "",
        "key": "",
        "pathname": "",
        "search": "",
        "state": Object {},
      }
    }
    match={
      Object {
        "isExact": true,
        "params": Object {},
        "path": "",
        "url": "",
      }
    }
    staticContext={Object {}}
  >
    Hallo
  </WithStyles(withRouter(Connect(Drawer)))>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Hallo",
      "classes": Object {
        "appBar": "withRouter-Connect-Drawer---appBar-3",
        "appFrame": "withRouter-Connect-Drawer---appFrame-2",
        "content": "withRouter-Connect-Drawer---content-9",
        "drawerPaper": "withRouter-Connect-Drawer---drawerPaper-8",
        "grow": "withRouter-Connect-Drawer---grow-5",
        "loadingIndicator": "withRouter-Connect-Drawer---loadingIndicator-10",
        "logo": "withRouter-Connect-Drawer---logo-6",
        "navIconHide": "withRouter-Connect-Drawer---navIconHide-7",
        "root": "withRouter-Connect-Drawer---root-1",
        "toolBar": "withRouter-Connect-Drawer---toolBar-4",
      },
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 2,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "",
          "pathname": "",
          "search": "",
          "state": Object {},
        },
        "push": [Function],
        "replace": [Function],
      },
      "location": Object {
        "hash": "",
        "key": "",
        "pathname": "",
        "search": "",
        "state": Object {},
      },
      "match": Object {
        "isExact": true,
        "params": Object {},
        "path": "",
        "url": "",
      },
      "staticContext": Object {},
    },
    "ref": null,
    "rendered": "Hallo",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Hallo",
        "classes": Object {
          "appBar": "withRouter-Connect-Drawer---appBar-3",
          "appFrame": "withRouter-Connect-Drawer---appFrame-2",
          "content": "withRouter-Connect-Drawer---content-9",
          "drawerPaper": "withRouter-Connect-Drawer---drawerPaper-8",
          "grow": "withRouter-Connect-Drawer---grow-5",
          "loadingIndicator": "withRouter-Connect-Drawer---loadingIndicator-10",
          "logo": "withRouter-Connect-Drawer---logo-6",
          "navIconHide": "withRouter-Connect-Drawer---navIconHide-7",
          "root": "withRouter-Connect-Drawer---root-1",
          "toolBar": "withRouter-Connect-Drawer---toolBar-4",
        },
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 2,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "",
            "pathname": "",
            "search": "",
            "state": Object {},
          },
          "push": [Function],
          "replace": [Function],
        },
        "location": Object {
          "hash": "",
          "key": "",
          "pathname": "",
          "search": "",
          "state": Object {},
        },
        "match": Object {
          "isExact": true,
          "params": Object {},
          "path": "",
          "url": "",
        },
        "staticContext": Object {},
      },
      "ref": null,
      "rendered": "Hallo",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
